src/js/angular/ttyg/controllers/ttyg-view.controller.js

```diff
@@ -272,38 +279,44 @@ function TTYGViewCtrl(
         if (!agentToEdit) {
             agentToEdit = TTYGContextService.getSelectedAgent();
         }
-        const agentFormModel = agentFormModelMapper(agentToEdit);
-        const activeRepositoryInfo = repositoryInfoMapper($repositories.getActiveRepositoryObject());
-        const options = {
-            templateUrl: 'js/angular/ttyg/templates/modal/agent-settings-modal.html',
-            controller: 'AgentSettingsModalController',
-            windowClass: 'agent-settings-modal',
-            resolve: {
-                dialogModel: function () {
-                    return new AgentSettingsModal(
-                        activeRepositoryInfo,
-                        $scope.activeRepositoryList,
-                        agentFormModel,
-                        AGENT_OPERATION.EDIT
-                    );
-                }
-            },
-            size: 'lg'
-        };
-        $uibModal.open(options).result.then(
-            // confirmed handler
-            (data) => {
-                TTYGService.editAgent(data)
-                    .then((updatedAgent) => {
-                        toastr.success($translate.instant("ttyg.agent.messages.agent_save_successfully", {agentName: updatedAgent.name}));
-                        const hasSelectedAgent = TTYGContextService.getSelectedAgent();
-                        if (hasSelectedAgent && data.id === hasSelectedAgent.id) {
-                            TTYGContextService.selectAgent(updatedAgent);
+        TTYGContextService.getDefaultAgent()
+            .then((agentDefaultValues) => {
+                const agentFormModel = agentFormModelMapper(agentToEdit, agentDefaultValues);
```

Comment:
I think it'd be better to call the ttyg.service directly and move the transformation there